src/main/scala/org/allenai/pipeline/Producer.scala

```diff
@@ -47,6 +47,36 @@ trait Producer[T] extends PipelineStep with CachingEnabled with Logging {
     returnValue
   }
 
+  /**
+   * Lift a function to producers. This gives you ability to fully customize the step info
+   */
+  def map[R](stepInfo: PipelineStepInfo, f: T => R): Producer[R] = {
```

Comment:
Love the idea.  Now question is: what does it do the the pipeline DAG diagram?  Ideally the producer you started with would be on there, and would be the parent of the Producer you got by calling .map(), which would also appear.  If you just copy the stepInfo with a new name, does that do this?  I don't imagine it would?